posthog/api/authentication.py

```diff
@@ -74,6 +74,26 @@ def create(self, validated_data: Dict[str, str]) -> Any:
         return user
 
 
+class LoginPrecheckSerializer(serializers.Serializer):
+    email = serializers.EmailField()
+
+    def to_representation(self, instance: Dict[str, str]) -> Dict[str, Any]:
+        return instance
+
+    def create(self, validated_data: Dict[str, str]) -> Any:
+        email = validated_data.get("email")
+        domain = email[email.index("@") + 1 :]
+        domain_query = (
+            OrganizationDomain.objects.verified_domains()
```

Comment:
FYI @guidoiaquinti here's the logic we discussed where we only use the domain to check (regardless of whether there's an account or not).